File: 20200105/javascript/124.test.js
const solution = n => f(n - 1);
const f = n => n < 3 ? '124'[n] : f(div(n, 3) - 1) + f(n % 3);
const div = (a, b) => parseInt(a / b);

// test --------------------

test('simple', () => {
  expect(solution(1)).toBe('1');
  expect(solution(2)).toBe('2');
  expect(solution(3)).toBe('4');
  expect(solution(4)).toBe('11');
  expect(solution(5)).toBe('12');
  expect(solution(6)).toBe('14');
  expect(solution(7)).toBe('21');
  expect(solution(8)).toBe('22');
  expect(solution(9)).toBe('24');
  expect(solution(10)).toBe('41');
  expect(solution(11)).toBe('42');
  expect(solution(12)).toBe('44');
  expect(solution(13)).toBe('111');
});

File: 20200105/python/124_test.py
def solution(n):
    if n <= 3:
        return '124'[n - 1]
    return solution((n - 1) // 3) + solution(n % 3)


# test ------------------------

def test_sample():
    assert solution(1) == '1'
    assert solution(2) == '2'
    assert solution(3) == '4'
    assert solution(4) == '11'
    assert solution(5) == '12'
    assert solution(6) == '14'
    assert solution(7) == '21'
    assert solution(8) == '22'
    assert solution(9) == '24'
    assert solution(10) == '41'
    assert solution(11) == '42'
    assert solution(12) == '44'
    assert solution(13) == '111'
